tests/test_catx.py

```diff
@@ -42,6 +100,38 @@ def catx(mlp_builder: NetworkBuilder, request: pytest.FixtureRequest = None) ->
     return catx
 
 
+@pytest.fixture
+def catx_init_with_extras(
+    custom_network_with_dropout_extras: Type[CustomHaikuNetwork],
+    jax_observations: JaxObservations,
+    epsilon: float,
+    key: PRNGKey,
+    request: pytest.FixtureRequest = None,
+) -> Tuple[CATX, CATXState]:
+    if not request:
+        action_min = 0.0
+        action_max = 1.0
+    else:
+        action_min = request.param[0]
+        action_max = request.param[1]
+
+    catx = CATX(
+        custom_network=custom_network_with_dropout_extras,
+        optimizer=optax.adam(learning_rate=0.01),
+        discretization_parameter=4,
+        bandwidth=1.5 / 4,
+        action_min=action_min,
+        action_max=action_max,
+    )
+
+    network_extras = {"dropout_rate": 0.2}
+    state = catx.init(
+        obs=jax_observations, epsilon=epsilon, key=key, network_extras=network_extras
+    )
+
+    return catx, state
```

Comment:
it seems to me that we are creating a lot of fixtures that are only used once. 
Could a function do the trick and replace all these fixtures above?